### ISSUE TITLE
Config settings is not overwritten when use cli command

### ISSUE THREAD

**Issue by szykov** [cypress-io/cypress#22430]
### Current behavior

Config settings is not overwritten when use cli command.
`npx cypress run --config videosFolder='vids',screenshotsFolder='pics'`

> (Screenshots)
>   -  C:\foo\cypress\screenshots\foo.cy.ts\bar (failed).png    (1280x720)
> 
>   (Video)
>   -  Started processing:  Compressing to 32 CRF
>   -  Finished processing: C:\foo\cypress\videos\foo.cy.ts.mp4    (2 seconds)

### Desired behavior

Used folders should be **vids** and **pics** instead.

> (Screenshots)
>   -  C:\foo\cypress\pics\foo.cy.ts\bar (failed).png    (1280x720)
> 
>   (Video)
>   -  Started processing:  Compressing to 32 CRF
>   -  Finished processing: C:\foo\cypress\vids\foo.cy.ts.mp4    (2 seconds)

### Test code to reproduce

`npx cypress run --config videosFolder='pics',screenshotsFolder='vids'`

cypress.config.ts
```
import { defineConfig } from 'cypress';

export default defineConfig({
	e2e: {
		screenshotsFolder: 'screenshots',
		videosFolder: 'videos'
	}
});
```

### Cypress Version

10.1

**Comment by lmiller1990** [comment]
Hey, thanks for the bug report - I'll look into this one for you, first thing next week.

**Comment by lmiller1990** [comment]
A little late but I did look into this and confirm it's definitely a bug in how we are resolving configuration precedence. Seems like a regression. I'll fix it.

**Comment by lmiller1990** [comment]
Related: https://github.com/cypress-io/cypress/issues/22245

**Comment by lmiller1990** [comment]
For ref, resolved order of config here: https://docs.cypress.io/guides/references/configuration#Resolved-Configuration

**Comment by fs-jake** [comment]
Hi there, commenting on this bug as well (just commented on #22245). We are running into a bug with this after Cypress v10 as well. We have a custom `url-config.js` file that is used to dynamically set the base URL for various environments. We were importing this file into our `commands.js` file and using the dynamic URL variable in one of our custom commands. Prior to Cypress v10, the variable was getting read in correctly. However, after Cypress v10, the variable is not getting the correct value I'm assuming due to a change in precedence.

**Comment by szykov** [comment]
@lmiller1990 
I know you're super-duper busy. 
But it would be nice to hear some news or possible release where this bug fix goes. 
Sorry to disturb, your work is important.

**Comment by lmiller1990** [comment]
Hi @szykov, I've been away for 1 month, but now I am back in action. Regarding this bug: has not been fixed yet. Is this something you'd like to try patching? If so, happy to link you some places in the code base to look at.

Also - any specific use case? I'm asking since there might be another way to work around it. Since `cypress.config` is a JS module now, you can do something like

```js
export default defineConfig({
  e2e: {
    videosFolder: process.env.FOO === "BAR" ? "some_folder" : "another_folder"
  }
})
```

Not a fix, but a good work around until there is a fix.

**Comment by szykov** [comment]
@lmiller1990 
Thanks for the update. I wish I could help but no time for now.
We use it in DevOps pipelines as env variables when running CLI.
Thanks.

**Comment by lmiller1990** [comment]
In that case the work-around in my above post should be fine - instead of passing the config like you are doing, you can just do a conditional inside `cypress.config.js` with `process.env`. Not ideal, but it will unblock you until we fix the bug.

**Comment by aalexeyenko** [comment]
Any chance for this issue to be fixed any time soon?

**Comment by lmiller1990** [comment]
Right now it works with Svelte + Vite - are you looking for some SvelteKit specific behavior you cannot accomplish just using the cypress/svelte mount function?

**Comment by lmiller1990** [comment]
I'm unable to prioritize this issue right now - are you interested in taking a try to fix? 

Alternatively https://github.com/cypress-io/cypress/issues/22430#issuecomment-1297794767 as a work around might be useful for you.

**Comment by jennifer-shehane** [comment]
I ran into this bug and commented on my experience here: https://github.com/cypress-io/cypress/issues/28332#issue-1993243797

**Comment by julcsi333** [comment]
It would be nice to have this documented somewhere, maybe around [here](https://docs.cypress.io/app/references/configuration#Overriding-Options). For example a precedence list, so the user would have an idea which configurations will be effective if they have multiple. (env variable, config file, cli option, ...). 
If it is documented somewhere other than here, I couldn't find it.
